lib/css/walkCssTokens.js

```diff
@@ -733,16 +747,25 @@ module.exports.eatWhitespace = (input, pos) => {
 };
 
 /**
+ * Do not pass commentCallbacks when the range has already been walked.
+ * This will cause comments to be collected multiple times.
  * @param {string} input input
  * @param {number} pos position
+ * @param {CssTokenCallbacks} callbacks callbacks
+ * @param {number} direction consume direction -1: reverse, 1: forward
  * @returns {number} position after whitespace and comments
  */
-module.exports.eatWhitespaceAndComments = (input, pos) => {
+module.exports.eatWhitespaceAndComments = (
+	input,
+	pos,
+	callbacks,
+	direction = 1
+) => {
```

Comment:
I think this is not a correct implementation, our parser is written according to the specification and we should almost not touch it, except adding callbacks